tenable/io/scans.py

```diff
@@ -577,6 +580,10 @@ def export(self, scan_id, *filters, **kw):
             params['history_id'] = self._check(
                 'history_id', kw['history_id'], int)
 
+        if 'scan_type' in kw:
+            params['type'] = self._check('type', kw['scan_type'], str,['web-app'])
+            scan_type_query = '?type={}'.format(params['type']) if 'type' in params else ''
```

Comment:
I know smashing query params onto the URL path works, however I think the right solution here is to extend _wait_for_download in base.py to support passing keywords.  Then using params={'type': 'web-app'} instead.  This allows requests to properly construct the Request object.

Reply:
Example change to to the base.py that should work:

```
diff --git a/tenable/io/base.py b/tenable/io/base.py
index be70974..e3b9bde 100644
--- a/tenable/io/base.py
+++ b/tenable/io/base.py
@@ -89,17 +89,17 @@ class TIOEndpoint(APIEndpoint):

         return resp

-    def _wait_for_download(self, path, resource, resource_id, file_id):
+    def _wait_for_download(self, path, resource, resource_id, file_id, **kw):
         '''
         A simple method to centralize waiting for an export to enter a
         completed state.  The initial request will be made and then we will
         recheck every two and a half seconds after that.  Once the status
         returns one of the completed states, we will return the status.
         '''
-        status = self._api.get(path).json()['status']
+        status = self._api.get(path, **kw).json()['status']
         while status not in ['error', 'ready']:
             time.sleep(2.5)
-            status = self._api.get(path).json()['status']
+            status = self._api.get(path, **kw).json()['status']

         # If the status that has been reported back is "error", then we will
         # need to throw the appropriate error back to the user.
```

Reply:
You're right - that is a better choice.